HTML/nested.html

```diff
@@ -0,0 +1,42 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Nested Classes Example</title>
+</head>
+<body>
+    <h1>Nested Classes Example</h1>
+    <div id="output"></div>
+
+    <script>
+        class OuterClass {
+            constructor(name) {
+                this.name = name;
+            }
+
+            greet() {
+                console.log(`Hello from ${this.name}`);
+            }
+
+            class InnerClass {
+                constructor(innerName) {
+                    this.innerName = innerName;
+                }
+
+                innerGreet() {
+                    console.log(`Hello from inner class, ${this.innerName}`);
+                }
+            }
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider moving InnerClass outside OuterClass</b></div>
<div id="fix">

Consider moving the `InnerClass` outside of `OuterClass`. Nested class definitions are not standard in JavaScript and may lead to unexpected behavior.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -13,19 +13,21 @@
          class OuterClass {
              constructor(name) {
                  this.name = name;
              }
 
              greet() {
                  console.log('Hello from ${this.name}');
              }
          }
 
 -        class InnerClass {
 +        class OuterClassInner {
              constructor(innerName) {
                  this.innerName = innerName;
              }
 
              innerGreet() {
                  console.log('Hello from inner class, ${this.innerName}');
              }
          }
 
          const outer = new OuterClass('Outer');
          outer.greet();
 +        const inner = new OuterClassInner('Inner');
 +        inner.innerGreet();
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/Amruta101998/DS-and-Algorithm/pull/21#issuecomment-2437737202>#10c142</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged